events/tests.py

```diff
@@ -87,11 +135,134 @@ def test_get_events_index(self):
         self.assertEqual(response.status_code, 200)
 
     def test_past_deadline(self):
-        event = Event.objects.create(title='Test event number 4', slug='test_past_deadline',
-                                     sign_up_deadline=timezone.now(),
-                                     author_id=self.member.id)
-        self.assertIsNotNone(event)
-        c = Client(enforce_csrf_checks=False)
-        response = c.post(reverse('events:detail', args=[event.slug]), {'user': 'person6', 'email': 'person6@test.com'})
+        self.event.sign_up_deadline = timezone.now()
+        self.event.save()
+        self.assertIsNotNone(self.event)
+        c = Client()
+        response = c.post(reverse('events:detail', args=[self.event.slug]), self.content)
         self.assertEqual(response.status_code, 403)
-        self.assertEqual(event.get_registrations().count(), 0)
+        self.assertEqual(self.event.get_registrations().count(), 0)
+
+    def test_event_detail_view_with_and_without_passcode(self):
+        self.event.passcode = 'secret'
+        self.event.save()
+        c = Client()
+        wrong, right = {'passcode': 'wrong'}, {'passcode': 'secret'}
+        incorrect = c.post(reverse('events:detail', args=[self.event.slug]), wrong)
+        self.assertEqual(incorrect.status_code, 200)
```

Comment:
Should this be changed to return 401

Can be done as a later issue (create and resolve this comment)